### ISSUE TITLE
Slow exit on Ubuntu 22.04

### ISSUE THREAD

**Issue by StefanBratanov** [TobiWo/eth-duties#44]
Hi, first of all I want to say I really like the tool and find it very useful when I need to restart my node or upgrade clients. Not sure if expected, but after I see when my next duty is and I click `Ctrl + C` to exit CLI, It takes ~15 seconds to exit. Is that expected? Logs below.

```
03-05-2023 08:30:21 - [INFO] Logging next interval...
03-05-2023 08:30:21 - [INFO] Validator xxxxxx has next ATTESTATION duty in: 02:49 min. (slot: xxxxxx)

^C03-05-2023 08:30:36 - [INFO] Happy staking. See you for next maintenance 🙂 !
```

**Comment by TobiWo** [comment]
Hi @StefanBratanov , thanks for reporting. I know the issue. That's because the tool only runs in a while loop and waits for the defined interval (`--interval`) for the next cycle :grimacing: . I will change that to asynchronous behavior in the future, so that the user interrupt is recognized immediately. It is not that high on my priority list so it may take some time.

**Comment by StefanBratanov** [comment]
> Hi @StefanBratanov , thanks for reporting. I know the issue. That's because the tool only runs in a while loop and waits for the defined interval (`--interval`) for the next cycle 😬 . I will change that to asynchronous behavior in the future, so that the user interrupt is recognized immediately. It is not that high on my priority list so it may take some time.

Sounds great. Thanks for the quick update on the issue. Will look out for new releases.

**Comment by TobiWo** [comment]
Hi @StefanBratanov, some update on that topic. I finished the code change and now waiting for review by @LuisNaldo7 . So this issue will be closed with https://github.com/TobiWo/eth-duties/pull/45 .
FYI: This will not trigger a new release as I want to add another feature for the next release. However, if you want to take advantage of the new termination behavior you can use the `latest` docker tag. So in other words once the PR is merged it will be available on `main` and thus via the `latest` docker tag.

**Comment by StefanBratanov** [comment]
> Hi @StefanBratanov, some update on that topic. I finished the code change and now waiting for review by @LuisNaldo7 . So this issue will be closed with #45 . FYI: This will not trigger a new release as I want to add another feature for the next release. However, if you want to take advantage of the new termination behavior you can use the `latest` docker tag. So in other words once the PR is merged it will be available on `main` and thus via the `latest` docker tag.

Thanks a lot. :) Sounds great! Thank you for the update.
